src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let backupObj = Object.assign({}, state);
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        backupObj = { ...backupObj };
+        Object.assign(backupObj, action.extraData);
+        break;
+      case 'removeProperties':
+        backupObj = { ...backupObj };
+
+        for (const key of action.keysToRemove) {
+          delete backupObj[key];
+        }
+        break;
+
+      case 'clear':
+        backupObj = {};
+    }
+
+    result.push(backupObj);
```

Comment:
so you don't need `backupObj = { ...backupObj };` on each case, but here you do this:
```suggestion
    result.push({ ...backupObj });
```
-> create a copy of the obj on-fly and push it

Reply:
Yes definitelly